docs/examples/driver_examples/calibrate_keithley.py

```diff
@@ -0,0 +1,215 @@
+            )
+        time.sleep(time_delay)
+
+        # Debug output
+        print(
+            f"{channel}.source.calibratev({v_range}, {src_Z}, {DMM_Z_rdg:.6e}, {src_FS:.6f}, {DMM_FS_rdg:.6f})"
+        )
+        print(
+            f"{channel}.measure.calibratev({v_range}, Z_rdg, {DMM_Z_rdg:.6e}, FS_rdg, {DMM_FS_rdg:.6f})"
+        )
+
+        # Start negative calibration:
+        smu.write(f"{channel}.cal.polarity = {channel}.CAL_NEGATIVE")
+        smu.write(f"{channel}.source.levelv = -{src_Z}")
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_ON")
+
+        # Measure negative zero voltage with SMU and DMM:
+        time.sleep(time_delay)
+        smu.write(f"Z_rdg = {channel}.measure.v()")
+        DMM_Z_rdg = dmm_param_volt()
+
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_OFF")
+        smu.write(f"{channel}.source.levelv = -{src_FS}")
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_ON")
+
+        # Measure negative full scale voltage with DMM:
+        time.sleep(time_delay)
+        smu.write(f"FS_rdg = {channel}.measure.v()")
+        DMM_FS_rdg = dmm_param_volt()
+
+        # Write negative v_range calibration to SMU:
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_OFF")
+        time.sleep(time_delay)
+        smu.write(
+            f"{channel}.source.calibratev(-{v_range}, -{src_Z}, {DMM_Z_rdg}, -{src_FS}, {DMM_FS_rdg})"
+        )
+        if sense_mode != "SENSE_CALA":
+            time.sleep(time_delay)
+            smu.write(
+                f"{channel}.measure.calibratev(-{v_range}, Z_rdg, {DMM_Z_rdg}, FS_rdg, {DMM_FS_rdg})"
+            )
+        time.sleep(time_delay)
+
+        # Debug output
+        print(
+            f"{channel}.source.calibratev(-{v_range}, -{src_Z}, {DMM_Z_rdg:.6e}, -{src_FS:.6f}, {DMM_FS_rdg:.6f})"
+        )
+        print(
+            f"{channel}.measure.calibratev(-{v_range}, Z_rdg, {DMM_Z_rdg:.6e}, FS_rdg, {DMM_FS_rdg:.6f})"
+        )
+
+        time.sleep(time_delay)
+
+        smu.write(f"{channel}.cal.polarity = {channel}.CAL_AUTO")
+
+        # Reset the smu to default state
+        smu.write(f"{channel}.source.levelv = {src_Z}")
+
+
+#%% Load instruments (the SMU is assumed to be imported)
+from qcodes.instrument_drivers.Keysight.Keysight_34470A_submodules import (
+    Keysight_34470A,
+)
+
+dmm = Keysight_34470A("dmm", "TCPIP0::10.164.54.211::inst0::INSTR")
+
+#%% Define current experiment
+load_or_create_experiment(
+    "MeasurementSetupDebug", context.devices[-1].name, load_last_duplicate=True
+)
+
+# %% calibrate both channels
+dmm.aperture_time(1.0)
+dmm.range(1)
+dmm.autozero("OFF")
+dmm.autorange("OFF")
+
+calibrate_keithley_smu_v(smu, dmm)
+
+smu.smua.volt(0)
+smu.smub.volt(0)
+
+#%% calibrate single channel in specific range
+
+dmm.aperture_time(1.0)
+dmm.range(1)
+dmm.autozero("OFF")
+dmm.autorange("OFF")
```

Comment:
is there are reason for these settings to be outside of the calibration function?

Reply:
fixed